### PR TITLE
Set Jira URL for Jira notifications globally rather than in alerts

### LINKED ISSUE #2297: Set Jira URL for Jira notifications globally rather than in alerts
### Current Behavior

The Jira URL is set in the destination field, for each alert.

However the Jira credentials are set globally, which is incoherent (different Jira instances would need to have the same user and password).

### Proposed Behavior

The Jira URL should be set globally.

It is expected that users will only ever need to send Dependency-Track alert notifications to a single Jira instance, hence it makes sense to configure the Jira URL globally rather than having to fill it in each alert.

The destination field then becomes the Jira project key.

### Checklist

- [X] I have read and understand the [contributing guidelines](https://github.com/DependencyTrack/dependency-track/blob/master/CONTRIBUTING.md#filing-issues)
- [X] I have checked the [existing issues](https://github.com/DependencyTrack/dependency-track/issues) for whether this enhancement was already requested

### PR DESCRIPTION
**Pull request by Mvld3r**
Signed-off-by: Mvld3r <Mvld3r@users.noreply.github.com>

### Description

Set Jira URL for Jira notifications globally rather than in alerts

### Addressed Issue

Fixes #2297.

### Checklist

- [x] I have read and understand the [contributing guidelines](https://github.com/DependencyTrack/dependency-track/blob/github-templates/CONTRIBUTING.md#pull-requests)
- [ ] This PR fixes a defect, and I have provided tests to verify that the fix is effective
- [x] This PR implements an enhancement, and I have provided tests to verify that it works as intended
- [ ] This PR introduces changes to the database model, and I have added corresponding [update logic](https://github.com/DependencyTrack/dependency-track/tree/master/src/main/java/org/dependencytrack/upgrade)
- [x] This PR introduces new or alters existing behavior, and I have updated the [documentation](https://github.com/DependencyTrack/dependency-track/tree/master/docs/_docs) accordingly

### CODE REVIEW

File: src/main/java/org/dependencytrack/notification/publisher/AbstractWebhookPublisher.java [hunk 1]
```diff
@@ -33,7 +33,7 @@
 public abstract class AbstractWebhookPublisher implements Publisher {
 
     public void publish(final String publisherName, final PebbleTemplate template, final Notification notification, final JsonObject config) {
-        final Logger logger = Logger.getLogger(this.getClass());
+        final Logger logger = Logger.getLogger(AbstractWebhookPublisher.class);
```
Comment (nscuro) [inline src/main/java/org/dependencytrack/notification/publisher/AbstractWebhookPublisher.java:36]: What's the reason for this change? The logger should refer to the class extending `AbstractWebHookPublisher` (e.g. `JiraPublisher`), not the abstract class itself.

File: src/main/java/org/dependencytrack/notification/publisher/AbstractWebhookPublisher.java [hunk 2]
```diff
@@ -55,8 +55,8 @@ public void publish(final String publisherName, final PebbleTemplate template, f
         try {
             credentials = getBasicAuthCredentials();
         } catch (PublisherException e) {
-            logger.warn("An error occurred during the retrieval of credentials needed for notification publication. Skipping notification");
-            return;
+            logger.warn("An error occurred during the retrieval of credentials needed for notification publication. Skipping notification", e);
+            throw e;
```
Comment (nscuro) [inline src/main/java/org/dependencytrack/notification/publisher/AbstractWebhookPublisher.java:59]: What's the reason for this behavior change (re-throwing the exception)?

File: src/main/java/org/dependencytrack/notification/publisher/AbstractWebhookPublisher.java (hunk 1 above)
Comment (Mvld3r) [reply src/main/java/org/dependencytrack/notification/publisher/AbstractWebhookPublisher.java:36]: You're right, I think it was a local change that I forgot to remove from my commit. I'll revert to the previous behaviour.

File: src/main/java/org/dependencytrack/notification/publisher/AbstractWebhookPublisher.java (hunk 2 above)
Comment (Mvld3r) [reply src/main/java/org/dependencytrack/notification/publisher/AbstractWebhookPublisher.java:59]: I feel that the exception stacktrace should be part of the log. Multiple exceptions can arise from `getBasicAuthCredentials()`, including exceptions resulting from the retrieval and decryption of the stored jira password, and it would help the user to pinpoint the issue.

As for rethrowing the exception or silently returning, I leave it up to you to decide.

This isn't strictly in relation to the PR title, but is on a related subject (jira publisher) and doesn't grant a separated PR imo.

Comment (nscuro) [reply src/main/java/org/dependencytrack/notification/publisher/AbstractWebhookPublisher.java:59]: I'm with you on the logging the stacktrace part, but re-throwing the exception may have some undesirable side effects. Best case the exception is just logged twice because it will be handled again by the "unhandled exception handler", worst case other operations may be negatively affected.

**Review by nscuro — APPROVED** [review]
Thanks @Mvld3r! 🙌
